datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -320,11 +317,14 @@ pub(crate) struct SortPreservingMergeStream {
     /// An id to uniquely identify the input stream batch
     next_batch_id: usize,
 
-    /// min heap for record comparison
-    min_heap: BinaryHeap<SortKeyCursor>,
+    /// Heap that yields [`SortKeyCursor`] in increasing order
+    heap: BinaryHeap<Reverse<SortKeyCursor>>,
 
     /// target batch size
     batch_size: usize,
+
+    /// row converter
+    row_converter: RowConverter,
```

Comment:
As I mentioned in https://github.com/apache/arrow-rs/pull/2593 I worry that the key comparisons for dictionaries will be incorrect if we see record batches with different dictionaries but use the same converter

Reply:
The way dictionaries are interned is specifically designed to handle this

Reply:
Great -- so we'll handle that discussion on https://github.com/apache/arrow-rs/pull/2593